tidb-binlog/pump_client/selector_test.go

```diff
@@ -0,0 +1,28 @@
+package client
+
+import (
+	. "github.com/pingcap/check"
+)
+
+type testSelectorSuite struct{}
+
+var _ = Suite(&testSelectorSuite{})
+
+func (t *testSelectorSuite) TestHashTS(c *C) {
+	const (
+		nPumps = 11
+		nTS    = 1024
+	)
+	counter := [nPumps]int{}
+	for i := 0; i < nTS; i++ {
+		counter[uint(hashTs(int64(i)))%nPumps]++
+	}
+	avg := nTS / 11
+	for i := 0; i < nPumps; i++ {
+		diff := counter[i] - avg
+		if diff < 0 {
+			diff = -diff
+		}
+		c.Assert(diff, Less, 300)
```

Comment:
```suggestion
		c.Assert(diff, Less, 9)
```

nTS / 11 = 93 so with 300 this test will always pass, making the test useless.

"9" is the standard deviation of the binomial distribution B(1024, 1/11). If you want a tight bound you could use "5" instead of "9".